.github/CODEOWNERS

```diff
@@ -1 +1,5 @@
 * @hashicorp/tf-editor-experience-engineers
+
+# release configuration
+/.release/                              @hashicorp/release-engineering
+/.github/workflows/build.yml            @hashicorp/release-engineering
```

Comment:
We can double check this, but my understanding of the [relevant guide part](https://hashicorp.atlassian.net/wiki/spaces/RELENG/pages/2307588257/CODEOWNERS) is that this is only necessary where there's assumed ownership over these files by the RelEng team - typically a case of a bigger team which has more dedicated RelEng staff who actually manage the files, but I'm not entirely sure this applies to us here.

Relatedly, the last section of the guide page talks about ownership expectations, which in 95% of the common cases (where external contributors generally don't understand and change these files) implies we are the owners (i.e. responsible for review & merge) so I'm not sure there's benefit in putting the RelEng team down as owner here.

Reply:
After consulting with RelEng, I also added our team as an owner for the release configuration.